core/trino-main/src/main/java/io/trino/connector/system/MaterializedViewSystemTable.java

```diff
@@ -0,0 +1,110 @@
+ * limitations under the License.
+ */
+package io.trino.connector.system;
+
+import io.trino.FullConnectorSession;
+import io.trino.Session;
+import io.trino.metadata.Metadata;
+import io.trino.metadata.QualifiedObjectName;
+import io.trino.security.AccessControl;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.connector.ConnectorTableMetadata;
+import io.trino.spi.connector.ConnectorTransactionHandle;
+import io.trino.spi.connector.InMemoryRecordSet;
+import io.trino.spi.connector.MaterializedViewFreshness;
+import io.trino.spi.connector.RecordCursor;
+import io.trino.spi.connector.SystemTable;
+import io.trino.spi.predicate.TupleDomain;
+
+import javax.inject.Inject;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.connector.system.jdbc.FilterUtil.tablePrefix;
+import static io.trino.metadata.MetadataListing.getMaterializedViews;
+import static io.trino.metadata.MetadataListing.listCatalogs;
+import static io.trino.spi.connector.SystemTable.Distribution.SINGLE_COORDINATOR;
+import static java.util.Objects.requireNonNull;
+
+public class MaterializedViewSystemTable
+        implements SystemTable
+{
+    private final Metadata metadata;
+    private final AccessControl accessControl;
+    private final MaterializedViewsSystemTableAdapter materializedViewAdapter;
+
+    @Inject
+    public MaterializedViewSystemTable(
+            Metadata metadata,
+            AccessControl accessControl,
+            MaterializedViewsSystemTableAdapter materializedViewAdapter)
+    {
+        this.metadata = requireNonNull(metadata, "metadata is null");
+        this.accessControl = requireNonNull(accessControl, "accessControl is null");
+        this.materializedViewAdapter = requireNonNull(materializedViewAdapter, "materializedViewAdapter is null");
+    }
+
+    @Override
+    public Distribution getDistribution()
+    {
+        return SINGLE_COORDINATOR;
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata()
+    {
+        return materializedViewAdapter.getTableMetadata();
+    }
+
+    @Override
+    public RecordCursor cursor(
+            ConnectorTransactionHandle transactionHandle,
+            ConnectorSession connectorSession,
+            TupleDomain<Integer> constraint)
+    {
+        Session session = ((FullConnectorSession) connectorSession).getSession();
+
+        List<Object[]> materializedViewRows = getAllMaterializedViews(session, constraint).stream()
+                .map(materializedViewAdapter::toTableRow)
+                .collect(toImmutableList());
+
+        InMemoryRecordSet.Builder displayTable = InMemoryRecordSet.builder(getTableMetadata());
+        materializedViewRows.forEach(displayTable::addRow);
+
+        return displayTable.build().cursor();
+    }
+
+    private List<MaterializedViewInfo> getAllMaterializedViews(Session session, TupleDomain<Integer> constraint)
```

Comment:
List -> Stream

Reply:
Changed to return void since display table is passed in to be filled with rows.